move ZModulo to test suite (#17258)
coq/coq: doc/changelog/11-standard-library/17258-remove-Zmodulo.rst

> - **Removed:** :g:`Numbers.Cyclic.ZModulo` from the standard library. This file
>   was deprecated in 8.17 and had no known use cases. The same code remains a
>   part of a test suite to ensurce consistency of :g:`CyclicAxioms`

Reviewer: ```suggestion
- **Removed:** :g:`Numbers.Cyclic.ZModulo` from the standard library. This file
  was deprecated in 8.17 and has no known use cases. It is retained in
  the test suite to ensure consistency of :g:`CyclicAxioms`
```

Please at least correct the typo "ensurce".

Author: Thank you!
